app/fragments/holders/HoldersLandingFragment.tsx

```diff
@@ -0,0 +1,329 @@
+import * as React from 'react';
+import { ActivityIndicator, Platform, Text, View, KeyboardAvoidingView, Alert, Pressable } from 'react-native';
+import WebView from 'react-native-webview';
+import Animated, { Easing, FadeIn, useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { WebViewMessageEvent, WebViewNavigation } from 'react-native-webview/lib/WebViewTypes';
+import { useTypedNavigation } from '../../utils/useTypedNavigation';
+import { t } from '../../i18n/t';
+import { useEngine } from '../../engine/Engine';
+import { warn } from '../../utils/log';
+import { AndroidToolbar } from '../../components/topbar/AndroidToolbar';
+import { StatusBar } from 'expo-status-bar';
+import { extractDomain } from '../../engine/utils/extractDomain';
+import { useParams } from '../../utils/useParams';
+import { HoldersAppParams } from './HoldersAppFragment';
+import { extractHoldersQueryParams } from './utils';
+import { getLocales } from 'react-native-localize';
+import { fragment } from '../../fragment';
+import { useAppConfig } from '../../utils/AppConfigContext';
+import { useKeysAuth } from '../../components/secure/AuthWalletKeys';
+import { OfflineWebView } from './components/OfflineWebView';
+import * as FileSystem from 'expo-file-system';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { normalizePath } from '../../engine/holders/HoldersProduct';
+
+export const HoldersLandingFragment = fragment(() => {
+    const { Theme } = useAppConfig();
+    const webRef = useRef<WebView>(null);
+    const authContext = useKeysAuth();
+    const engine = useEngine();
+    const navigation = useTypedNavigation();
+    const [hideKeyboardAccessoryView, setHideKeyboardAccessoryView] = useState(true);
+    const { endpoint, onEnrollType } = useParams<{ endpoint: string, onEnrollType: HoldersAppParams }>();
+    const lang = getLocales()[0].languageCode;
+    const currency = engine.products.price.usePrimaryCurrency();
+    const [offlineAppReady, setOfflineAppReady] = useState<{ version: string } | false>();
+    useEffect(() => {
```

Comment:
Do we really need this and in asynchronous way?

Reply:
how would you check files that are sync read from FileSystem insted?

Reply:
yeah, rewrote just using current stored version, this previously was made to check resources because those were stored in cache directory (and OS could delete those at any time)